### PR TITLE
Feature/array converte generator

### PR DESCRIPTION
**Pull request by roman-kozachenko**

### CODE REVIEW

File: src/msgpack.light/Converters/Generation/DuplicateArrayElementException.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+
+namespace ProGaudi.MsgPack.Light.Converters.Generation
+{
+    public class DuplicateArrayElementException : Exception
```
Comment (aensidhe) [inline src/msgpack.light/Converters/Generation/DuplicateArrayElementException.cs]: Let's create and use GeneratorException derived from some SerializationException

File: tests/msgpack.light.tests/Generators/ClassInheritanceTest.cs [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
-
-        public ClassInheritanceTest(Fixture fixture)
-        {
-            _fixture = fixture;
-        }
-
-        [Fact]
-        public void WriteSmoke()
+        [Theory]
+        [ClassData(typeof(FixtureProvider))]
```
Comment (aensidhe) [inline tests/msgpack.light.tests/Generators/ClassInheritanceTest.cs]: The idea was to provide one time initialization for context. Why should we drop it?

**Review by aensidhe — APPROVED** [review]

File: src/msgpack.light/Converters/Generation/DuplicateArrayElementException.cs (hunk 1 above)
Comment (roman-kozachenko) [reply src/msgpack.light/Converters/Generation/DuplicateArrayElementException.cs]: Common class for Duplication*ElementException created. Exception creation extracted to ExceptionUtils.

File: tests/msgpack.light.tests/Generators/ClassInheritanceTest.cs (hunk 2 above)
Comment (roman-kozachenko) [reply tests/msgpack.light.tests/Generators/ClassInheritanceTest.cs]: One-time initialziation is performed using static constuctors.

**Pull request merged**